### PR TITLE
DEV: Add `above-topic-list-item` plugin outlet and pass topic in `above-latest-topic-list-item` args

### PR DESCRIPTION
**Pull request by OsamaSayegh**
This PR does a couple of things:

1. Add a new plugin outlet, `above-topic-list-item`, to the `topic-list-item` component
2. Pass the topic in question as an outlet argument for the (existing) `above-latest-topic-list-item` outlet in the `latest-topic-list-item` component.

### CODE REVIEW

**Review by tgxworld — DISMISSED** [review]

File: app/assets/javascripts/discourse/app/components/topic-list-item.hbs [hunk 1]
```diff
@@ -1 +1,5 @@
+<PluginOutlet
+  @name="above-topic-list-item"
+  @outletArgs={{hash topic=this.topic}}
+/>
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/components/topic-list-item.hbs:4]: Actually I'm wondering if we can make use of the new plugin outlet feature here to register click handlers? Can you look into the possibility of that?

Comment (OsamaSayegh) [reply app/assets/javascripts/discourse/app/components/topic-list-item.hbs:4]: Can you clarify/remind me of which feature you're thinking of here?

Comment (tgxworld) [reply app/assets/javascripts/discourse/app/components/topic-list-item.hbs:4]: Hmm two weeks ago I was thinking of using a plugin outlet but I don't think that will work. By the way do we need this outlet considering that we already have 

https://github.com/discourse/discourse/blob/00b1b88a86aa2837e53276c5af805f57f7f183de/app/assets/javascripts/discourse/app/raw-templates/list/topic-list-item.hbr#L1

Comment (OsamaSayegh) [reply app/assets/javascripts/discourse/app/components/topic-list-item.hbs:4]: > By the way do we need this outlet considering that we already have

That's a raw template which doesn't support modifiers such as `did-insert` and `will-destroy` which I need for the use case in https://github.com/discourse/discourse-clickable-topic/pull/4. Moreover, we're going to remove raw templates in the near future so this new outlet will probably be a perfect replacement for the raw one that you mentioned.

**Review by tgxworld — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/topic-list-item.hbs (hunk 1 above)
Comment (tgxworld) [reply app/assets/javascripts/discourse/app/components/topic-list-item.hbs:4]: Ic. Thank you for the clarifications 👍

**Pull request merged**